lib/internal/event_target.js

```diff
@@ -973,6 +834,98 @@ ObjectDefineProperties(NodeEventTarget.prototype, {
 
 // EventTarget API
 
+/**
+ * @see https://dom.spec.whatwg.org/#dictdef-eventlisteneroptions
+ * @typedef {Object} AddEventListenerOptions
+ * @property {null} __proto__
+ * @property {boolean} [capture=false]
+ * @property {boolean} passive
+ * @property {boolean} [once=false]
+ * @property {import('./abort_controller').AbortSignal|null} signal
+ */
+
+/**
+ * @see https://dom.spec.whatwg.org/#event-flatten-more
+ * @param {*} options
+ * @returns {AddEventListenerOptions}
+ */
+function flattenMoreOptions (options) {
+  if (typeof options === 'boolean') {
+    return {
+      __proto__: null,
+      capture: options,
+      passive: false,
+      once: false,
+      signal: null 
+    }
+  }
+
+  validateDictionary(options, 'options')
+
+  const validatedOptions = {
+    __proto__: null,
+    passive: Boolean(options.passive),
+    once: Boolean(options.once),
+    capture: Boolean(options.capture),
+    signal: null
+  }
+  
+  const signal = options.signal // only hit getter once
+  if (signal != null)
+    validateAbortSignal(signal, 'signal')
+
+  validatedOptions.signal = signal
+
+  return validatedOptions
+}
+
+/**
+ * @see https://dom.spec.whatwg.org/#add-an-event-listener
+ * @param {EventTarget} eventTarget
+ * @param {EventListenerStruct} listener
+ */
+function addAnEventListener (eventTarget, listener) {
+  if (listener.signal != null && listener.signal.aborted) {
+    return
+  }
+
+  if (listener.callback == null) {
+    return
+  }
+
+  listener.passive ??= false
+
+  const event = eventTarget.kTestIndices.get(listener.type)
+
+  if (!event) {
+    eventTarget.kTestIndices.set(listener.type, [listener])
+  } else if (
+    !ArrayPrototypeSome(event, (value) => {
+      return value.callback === listener.callback && value.capture === listener.capture
+    })
+  ) {
+    ArrayPrototypePush(event, listener)
+    eventTarget.kTestIndices.set(listener.type, event)
+  }
+
+  if (listener.signal) {
+    listener.signal.addEventListener('abort', () => {
```

Comment:
I recommend avoiding if statements / branches as much as possible 

```suggestion
    listener.signal?.addEventListener('abort', () => {
```